### DISCUSSION TITLE
Problems with the CLS value at Lighthouse test

### DISCUSSION THREAD

**Discussion by Miningmark** [neuefische/questions#852] [Web] [answered]
Hello,
For a few days now I've been trying to solve the problem of the extremely high CLS value. In the meantime I've tried a few things, including making the header less dynamic, doing without the theme provider of styled components, splitting the header into mobile and desktop and hiding it with conditinal rendering or with CSS “display none”. Unfortunately, all these approaches have not helped and the CLS value continues to fluctuate between approx. 0.7-1.2. It cannot be due to the images because this problem also occurs when you are on the 404 or impressum page, for example, where only text can be found.

Repository: https://github.com/Miningmark/yumekai-website
Vercel link: https://yumekai-website.vercel.app/404

![grafik](https://github.com/user-attachments/assets/cd3f7306-4d99-4695-a324-c6227aac3797)

**Comment by onemanwenttomow** [comment, marked as answer]
Hi @Miningmark,

The issue seems to be that the CSS is only rendering on the client. (If you watch closely on the page you will see a "flash", or if you look at the snapshots in the google lighthouse report you can also sometimes see it). 

A possible solution would be to use [`ServerStyleSheet`](https://styled-components.com/docs/advanced#server-side-rendering).

Here is some boiler-plate (that we use in our other recap projects) that you could tailor to your needs (for `_document.js`)

``` js
import Document, { Html, Head, Main, NextScript } from "next/document";
import { ServerStyleSheet } from "styled-components";

export default class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const sheet = new ServerStyleSheet();
    const originalRenderPage = ctx.renderPage;

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) =>
            sheet.collectStyles(<App {...props} />),
        });

      const initialProps = await Document.getInitialProps(ctx);
      return {
        ...initialProps,
        styles: [initialProps.styles, sheet.getStyleElement()],
      };
    } finally {
      sheet.seal();
    }
  }

  render() {
    return (
      <Html lang="en">
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

```

I hope this helps.

**Reply by Miningmark** [reply]
Thanks Pete for this solution. I've tried so much that I didn't even notice it anymore :)
